sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -272,18 +259,15 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
       schema: StructType,
       options: Map[String, String]): DataFrame = {
     val tableIdent = sparkSession.sessionState.sqlParser.parseTableIdentifier(tableName)
-    val cmd =
-      CreateTableUsing(
-        tableIdent,
-        userSpecifiedSchema = Some(schema),
-        source,
-        temporary = false,
-        options,
-        partitionColumns = Array.empty[String],
-        bucketSpec = None,
-        allowExisting = false,
-        managedIfNoPath = false)
-    sparkSession.sessionState.executePlan(cmd).toRdd
+    val tableDesc = CatalogTable(
+      identifier = tableIdent,
+      tableType = CatalogTableType.EXTERNAL,
+      storage = CatalogStorageFormat.empty.copy(properties = options),
+      schema = schema,
+      provider = Some(source)
+    )
+    val plan = CreateTable(tableDesc, SaveMode.ErrorIfExists, None)
```

Comment:
This change will have an external impact. The value of source is input from users. That means it could be "hive". That means, users can create a Hive serde Table. If this is desired, we need to document it and test it?